src/Tickets/Commerce/Cart/Agnostic_Cart.php

```diff
@@ -0,0 +1,525 @@
+	 *
+	 * @param string $item_id    The item ID.
+	 * @param int    $quantity   The quantity to update.
+	 * @param ?array $extra_data Extra data to save to the item.
+	 *
+	 * @return void
+	 */
+	protected function update_item( $item_id, int $quantity, ?array $extra_data = null ): void {
+		$item_object  = $this->items[ $item_id ];
+		$new_quantity = $item_object->add_quantity( $quantity );
+
+		// If the quantity is less than 1, remove the item from the cart.
+		if ( $new_quantity < 1 ) {
+			$this->remove_item( $item_id );
+
+			return;
+		}
+
+		// Maybe update the extra data.
+		if ( null !== $extra_data ) {
+			$item_object['extra'] = $extra_data;
+		}
+
+		$this->reset_calculations();
+	}
+
+	/**
+	 * Removes an item from the cart.
+	 *
+	 * @since TBD
+	 *
+	 * @param int|string $item_id  The item ID.
+	 * @param null|int   $quantity The quantity to remove.
+	 */
+	public function remove_item( $item_id, $quantity = null ) {
+		if ( null === $quantity ) {
+			unset( $this->items[ $item_id ] );
+			$this->reset_calculations();
+
+			return;
+		}
+
+		$this->update_item( $item_id, -abs( (int) $quantity ) );
+	}
+
+	/**
+	 * Process the items in the cart.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $data to be processed by the cart.
+	 *
+	 * @return array|bool
+	 */
+	public function process( array $data = [] ) {
+		if ( empty( $data ) ) {
+			return false;
+		}
+
+		// Reset the contents of the cart.
+		$this->clear();
+
+		/** @var Tickets_Handler $tickets_handler */
+		$tickets_handler = tribe( 'tickets.handler' );
+
+		// Prepare the error message array.
+		$errors = [];
+
+		// Natively handle adding tickets as items to the cart.
+		foreach ( $data['tickets'] as $ticket ) {
+			// Enforces that the min to add is 1.
+			$quantity = max( 1, (int) $ticket['quantity'] );
+
+			// Check if the ticket can be added to the cart.
+			$can_add_to_cart = $tickets_handler->ticket_has_capacity( $ticket['ticket_id'], $quantity, $ticket['obj'] );
+
+			// Skip and add to the errors if the ticket can't be added to the cart.
+			if ( is_wp_error( $can_add_to_cart ) ) {
+				$errors[] = $can_add_to_cart;
+
+				continue;
+			}
+
+			// Add to / update quantity in cart.
+			$this->add_item( $ticket['ticket_id'], $quantity, $ticket['extra'] );
+		}
+
+		/**
+		 * Fires after the ticket data has been processed.
+		 *
+		 * This allows for further processing of data within the $data array.
+		 *
+		 * @since TBD
+		 *
+		 * @param Cart_Interface $cart The cart object.
+		 * @param array          $data The data to be processed by the cart.
+		 */
+		do_action( 'tec_tickets_commerce_cart_process', $this, $data );
+
+		// Saved added items to the cart.
+		$this->save();
+
+		if ( ! empty( $errors ) ) {
+			return $errors;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Add the full set of parameters to the items in the cart.
+	 *
+	 * @param array $items The items in the cart.
+	 *
+	 * @return array The items in the cart with the full set of parameters.
+	 */
```

Comment:
```suggestion
	/**
	 * Add the full set of parameters to the items in the cart.
	 *
	 * @since TBD
	 *
	 * @param array $items The items in the cart.
	 *
	 * @return array The items in the cart with the full set of parameters.
	 */
```